src/main/java/no/finn/unleash/util/UnleashConfig.java

```diff
@@ -1,491 +1,9 @@
-            @Nullable String namePrefix,
-            long fetchTogglesInterval,
-            long sendMetricsInterval,
-            boolean disableMetrics,
-            UnleashContextProvider contextProvider,
-            boolean isProxyAuthenticationByJvmProperties,
-            boolean synchronousFetchOnInitialisation,
-            @Nullable UnleashScheduledExecutor unleashScheduledExecutor,
-            @Nullable UnleashSubscriber unleashSubscriber,
-            @Nullable Strategy fallbackStrategy,
-            @Nullable ToggleBootstrapProvider unleashBootstrapProvider,
-            @Nullable Proxy proxy,
-            @Nullable Authenticator proxyAuthenticator) {
-
-        if (appName == null) {
-            throw new IllegalStateException("You are required to specify the unleash appName");
-        }
-
-        if (instanceId == null) {
-            throw new IllegalStateException("You are required to specify the unleash instanceId");
-        }
-
-        if (unleashAPI == null) {
-            throw new IllegalStateException("You are required to specify the unleashAPI url");
-        }
-
-        if (unleashScheduledExecutor == null) {
-            throw new IllegalStateException("You are required to specify a scheduler");
-        }
-
-        if (unleashSubscriber == null) {
-            throw new IllegalStateException("You are required to specify a subscriber");
-        }
-
-        if (fallbackStrategy == null) {
-            this.fallbackStrategy = UNKNOWN_STRATEGY;
-        } else {
-            this.fallbackStrategy = fallbackStrategy;
-        }
-
-        if (isProxyAuthenticationByJvmProperties && proxyAuthenticator == null) {
-            enableProxyAuthentication();
-        } else if (proxyAuthenticator != null) {
-            Authenticator.setDefault(proxyAuthenticator);
-        }
-
-        this.unleashAPI = unleashAPI;
-        this.customHttpHeaders = customHttpHeaders;
-        this.customHttpHeadersProvider = customHttpHeadersProvider;
-        this.unleashURLs = new UnleashURLs(unleashAPI);
-        this.appName = appName;
-        this.environment = environment;
-        this.instanceId = instanceId;
-        this.sdkVersion = sdkVersion;
-        this.backupFile = backupFile;
-        this.projectName = projectName;
-        this.namePrefix = namePrefix;
-        this.fetchTogglesInterval = fetchTogglesInterval;
-        this.sendMetricsInterval = sendMetricsInterval;
-        this.disableMetrics = disableMetrics;
-        this.contextProvider = contextProvider;
-        this.isProxyAuthenticationByJvmProperties = isProxyAuthenticationByJvmProperties;
-        this.synchronousFetchOnInitialisation = synchronousFetchOnInitialisation;
-        this.unleashScheduledExecutor = unleashScheduledExecutor;
-        this.unleashSubscriber = unleashSubscriber;
-        this.toggleBootstrapProvider = unleashBootstrapProvider;
-        this.proxy = proxy;
-    }
-
-    public static Builder builder() {
```

Comment:
Should this method remain and return the new builder?

Reply:
Since we're extending, the method is available and returns the new builder. I tested

Reply:
Oops! You are right!